cli/src/commands/extensions/extension.rs

```diff
@@ -0,0 +1,167 @@
+use std::convert::TryFrom;
+use std::fs::{self, Permissions};
+#[cfg(unix)]
+use std::os::unix::prelude::PermissionsExt;
+use std::path::PathBuf;
+
+use anyhow::{anyhow, Result};
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde::Deserialize;
+use walkdir::WalkDir;
+
+const MANIFEST_NAME: &str = "PhylumExt.toml";
+
+lazy_static! {
+    static ref EXTENSION_NAME_RE: Regex = Regex::new(r#"^[a-z0-9_-]+$"#).unwrap();
```

Comment:
It's a safe default which makes things easy to check and enforces a consistent styling to a degree, solving at the same time conflict issues in contexts like case-insensitive filesystems (i.e. Windows).

Reply:
Did you take inspiration from other tools like GitHub's CLI, or did you just pick the minimal possible set of characters you deemed necessary?

Reply:
The latter -- I think this does not impose an uncomfortable restriction, as e.g. most package managers follow that pattern, on the users and is at the same time easy to check for. If we can come up with compelling arguments for doing otherwise (being less or even _more_ restrictive, like say [npm](https://github.com/npm/validate-npm-package-name#naming-rules)), I'm not opposed to changing this.

Reply:
Having the first character be a letter is likely a prerequisite to not have clap yell at us, so that's a great point.